pkg/sql/sequence.go

```diff
@@ -182,25 +183,48 @@ func (p *planner) incrementSequenceUsingCache(
 
 		// This sequence has exceeded its bounds after performing this increment.
 		if endValue > seqOpts.MaxValue || endValue < seqOpts.MinValue {
-			// If the sequence exceeded its bounds prior to the increment, then return an error.
-			if (seqOpts.Increment > 0 && endValue-seqOpts.Increment*(cacheSize-1) > seqOpts.MaxValue) ||
-				(seqOpts.Increment < 0 && endValue-seqOpts.Increment*(cacheSize-1) < seqOpts.MinValue) {
-				return 0, 0, 0, boundsExceededError(descriptor)
+
+			abs := func(i int64) int64 {
+				if i < 0 {
+					return -i
+				}
+				return i
 			}
+
 			// Otherwise, values between the limit and the value prior to incrementing can be cached.
 			limit := seqOpts.MaxValue
 			if seqOpts.Increment < 0 {
 				limit = seqOpts.MinValue
 			}
-			abs := func(i int64) int64 {
-				if i < 0 {
-					return -i
+
+			// TODO
+			// if cache size is 1 then it fails?
+			// start value is inc * cache
+			// value should be set to the cache -1
+			startValue := endValue - increaseValue
+			diff := limit - startValue
+			sizeOfCache = abs(diff / seqOpts.Increment)
+			lastValue := startValue + seqOpts.Increment*(sizeOfCache)
+
+			// Set the sequence to the limit so that next fetchNextValues is not over extended.
+			if createdInCurrentTxn {
+				if err = p.txn.Put(ctx, seqValueKey, lastValue); err != nil {
+					return 0, 0, 0, err
+				}
+			} else {
+				if err = p.ExecCfg().DB.Put(ctx, seqValueKey, lastValue); err != nil {
+					return 0, 0, 0, err
```

Comment:
This is not good because what if it doesn't happen but should.

Reply:
Im going to look into ways such as cput? still not fully aware but i think going into inc is a bit dangerous